### PR TITLE
Streamlined presets, better libs for programattic use

### PR DESCRIPTION
**Pull request by jleeh**
Added generic ability to override config on deployment, for example:
```
NAMESPACE_PREFIX=chainlink-the-best
CHARTS='{"chainlink": {"values": {"chainlink": {"image": {"version": "0.10.15"}}}}}'
```

### CODE REVIEW

File: environment/chart.go [hunk 1]
```diff
@@ -37,86 +36,66 @@ const (
...
+	ChartConnections ChartConnections       `json:"chart_connections,omitempty"`
 }
 
 // HelmChart helm chart structure
 type HelmChart struct {
 	Name          string
 	NamespaceName string
-	settings      *ChartSettings
+
+	settings      *Chart
```
Comment (kalverra) [inline environment/chart.go:52]: Make more sense to be named just `chart` instead of `settings` then?

**Review by kalverra — CHANGES_REQUESTED** [review]
Just some naming questions/considerations.

File: cmd/cli/envcli.go [hunk 2]
```diff
@@ -5,66 +5,66 @@ import (
...
-				Required: false,
-			},
-			&cli.StringFlag{
-				Name:     "preset",
-				Aliases:  []string{"p"},
-				Usage:    "environment preset name",
-				Required: true,
-			},
-		},
-		Before: func(c *cli.Context) error {
```
Comment (skudasov) [inline cmd/cli/envcli.go:36]: Is there any other check that completely prevents a user to interact with an env when both `persistent` and `persistent_connection` are set to false?

File: cmd/cli/envcli.go [hunk 3]
```diff
@@ -5,66 +5,66 @@ import (
...
 				Action: func(c *cli.Context) error {
-					presetPath := c.String("preset")
-					e, err := environment.LoadEnvironment(presetPath)
+					environmentPath := c.String("environment")
+					e, err := environment.LoadPersistentEnvironment(environmentPath)
 					if err != nil {
 						return err
 					}
-					if err := e.Connect(); err != nil {
+					if !e.Config.PersistentConnection {
```
Comment (skudasov) [inline cmd/cli/envcli.go:64]: I think it's better to forbid every possible interaction when using CLI (Debugging), so we can have only two modes:
1. CI run with `false/false` 
2. Local run where you setup env yourself and do that only once.
Local run will save some resources when debugging and make you write tests more carefully with, for example, UUIDs and cleanup, so if we decide to "batch" the tests later we don't need to "fix" them again.

**Review by skudasov — APPROVED** [review]
Great job!

**Pull request merged**